### PR TITLE
Add a trait to make defining mock responses simpler

### PR DESCRIPTION
**Pull request by markstory**
The trait automatically clears up mocks and makes possible to define mocks with no static class name references.

### CODE REVIEW

File: tests/TestCase/TestSuite/HttpClientTraitTest.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+use Cake\TestSuite\TestCase;
+
+class HttpClientTraitTest extends TestCase
+{
+    use HttpClientTrait;
+
+    /**
+     * Provider for http methods.
+     *
+     * @return void
```
Comment (othercorey) [inline tests/TestCase/TestSuite/HttpClientTraitTest.php:30]: returns array<array<string>> ?

File: src/TestSuite/HttpClientTrait.php [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    /**
+     * Create a new response.
+     *
+     * @param int $code The response code to use. Defaults to 200
+     * @param array $headers A list of headers for the response. Example `Content-Type: application/json`
+     * @param string $body The body for the response.
+     * @return \Cake\Http\Client\Response
+     */
+    public function newResponse(int $code = 200, array $headers = [], string $body = ''): Response
```
Comment (othercorey) [inline src/TestSuite/HttpClientTrait.php:46]: Should this have `Client` in the method name to match the others?

File: tests/TestCase/TestSuite/HttpClientTraitTest.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            ['Patch'],
+            ['Delete'],
+        ];
+    }
+
+    /**
+     * @dataProvider methodProvider
+     * @param string $httpMethod Http Method name.
+     */
+    public function testRequestMethods($httpMethod)
```
Comment (othercorey) [inline tests/TestCase/TestSuite/HttpClientTraitTest.php:47]:
```suggestion
     */
    public function testRequestMethods(string $httpMethod)
```

just type hint it

File: tests/TestCase/TestSuite/HttpClientTraitTest.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        $traitMethod = "mockClient{$httpMethod}";
+        $clientMethod = strtolower($httpMethod);
```
Comment (othercorey) [inline tests/TestCase/TestSuite/HttpClientTraitTest.php:50]: what does strtolower do?

Comment (markstory) [reply tests/TestCase/TestSuite/HttpClientTraitTest.php:49]: Gets the casing to match the method names on the http client.

Comment (othercorey) [reply tests/TestCase/TestSuite/HttpClientTraitTest.php:49]: Method names are case insensitive.

Comment (markstory) [reply tests/TestCase/TestSuite/HttpClientTraitTest.php:49]: True, I can remove the case change.

**Pull request merged**